### PR TITLE
JDK-8295088: Update External Spec page to show tabs for hosts

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a medium but interesting fix to display tabs on the `External Specifications` page to indicate the host name for the spec.

The underlying "fun" part is to update `Table` to be `Table<T>`, where `T` is the type parameter for the predicates used to classify each row to determine the tabs when it will be displayed.

The changes to `Table` are relatively simple, but there is a downstream cost to update all the existing use sites to avoid introducing any raw types (i.e. because `Table` now takes a type parameter).   While it would have been simple to just change existing uses from `Table` to `Table<Element>`, the changes are done more appropriately. In particular, if a table does not use tabs at all, the type parameter can be `Void`, and in other situations, the type parameter can be a subtype of `Element`, such as `ModuleElement`, `PackageElement` or `TypeElement`.

With `Table` updated to be `Table<T>`, the table in `ExternalSpecsWriter` can be `Table<URI>` with table-tab predicates examining the host name for the URL.

Eventually, it might be good to do a further upgrade to use a horizontal list of checkboxes instead of table tabs, for the potentially open-ended set of host names, but this will require an update to the JavaScript we use for checkboxes, which seems out of scope for the work presented here.   I note that in demo JDK docs, there are only about half-a-dozen spec hosts, so table tabs are a reasonable compromise for now.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295088](https://bugs.openjdk.org/browse/JDK-8295088): Update External Spec page to show tabs for hosts

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.org/census#hannesw) (@hns - **Reviewer**) ⚠️ Review applies to [a3af7eeb](https://git.openjdk.org/jdk/pull/10636/files/a3af7eebe40a877daa63278e6bf052e12d961d2c)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10636/head:pull/10636` \
`$ git checkout pull/10636`

Update a local copy of the PR: \
`$ git checkout pull/10636` \
`$ git pull https://git.openjdk.org/jdk pull/10636/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10636`

View PR using the GUI difftool: \
`$ git pr show -t 10636`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10636.diff">https://git.openjdk.org/jdk/pull/10636.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ModuleIndexWriter.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ protected void addIndex(Content target) {
 
         if (!groupModuleMap.keySet().isEmpty()) {
             TableHeader tableHeader = new TableHeader(contents.moduleLabel, contents.descriptionLabel);
-            Table table =  new Table(HtmlStyle.summaryTable)
+            var table =  new Table<ModuleElement>(HtmlStyle.summaryTable)
```
Comment (turbanoff) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ModuleIndexWriter.java:87]: let's remove one redundant space
```suggestion
            var table = new Table<ModuleElement>(HtmlStyle.summaryTable)
```

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/PackageIndexWriter.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ protected void addIndex(Content target) {
                 = configuration.group.groupPackages(packages);
 
         if (!groupPackageMap.keySet().isEmpty()) {
-            Table table =  new Table(HtmlStyle.summaryTable)
+            var table =  new Table<PackageElement>(HtmlStyle.summaryTable)
```
Comment (turbanoff) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/PackageIndexWriter.java:88]: Let's remove one redundant space
```suggestion
            var table = new Table<PackageElement>(HtmlStyle.summaryTable)
```

**Review by hns — APPROVED** [review]
Very nice work, Jon! Looks good to me.

**Pull request closed without merging**